packages/jazz-tools/src/schema/coMap.ts

```diff
@@ -0,0 +1,364 @@
+import { CoValueUniqueness, JsonValue, RawCoMap } from "cojson";
+import { Account, CoMap, Group } from "../exports.js";
+import {
+  CoValue,
+  CoValueClass,
+  ID,
+  IfCo,
+  ItemsSym,
+  RefsToResolve,
+  RefsToResolveStrict,
+  Resolved,
+  Schema,
+  SchemaInit,
+  UnCo,
+  co,
+  isRefEncoded,
+  parseCoValueCreateOptions,
+} from "../internal.js";
+import { CoValuePrototype } from "./interfaces.js";
+import { CoValueResolutionNode, ensureCoValueLoaded } from "./subscribe.js";
+export type Simplify<A> = {
+  [K in keyof A]: A[K];
+} extends infer B
+  ? B
+  : never;
+
+export class CoMapSchema {
```

Comment:
Decided to separate Schema and Instance for two resons:
1. It will be easier to adopt another Schema system
2. Has made the instance management easier to implement

This approach has broken all the Resolve types, so I've just turned off most of the type validations.